sbndqm/Archiver/Archiver.py

```diff
@@ -11,167 +11,288 @@
+                            TotalSetTime += SetLatest(r, stream_name, MetricDict[stream_name][0])
+                            TotalSetN += 1
+                            WritePostgres(p, cur, Config[stream_name][4], Config[stream_name][2], MetricDict[stream_name][3], int(MetricDict[stream_name][0].split('-')[0]))
+                            MetricDict[stream_name] = []
+                        #Case for "max" averaging.
+                        elif Config[stream_name][3] == 3:
+                            TotalSetTime += SetLatest(r, stream_name, MetricDict[stream_name][0])
+                            TotalSetN += 1
+                            WritePostgres(p, cur, Config[stream_name][4], Config[stream_name][2], MetricDict[stream_name][1], int(MetricDict[stream_name][0].split('-')[0]))
+                            MetricDict[stream_name] = []
+                        #Case for "min" averaging.
+                        elif Config[stream_name][3] == 4:
+                            TotalSetTime += SetLatest(r, stream_name, MetricDict[stream_name][0])
+                            TotalSetN += 1
+                            WritePostgres(p, cur, Config[stream_name][4], Config[stream_name][2], MetricDict[stream_name][1], int(MetricDict[stream_name][0].split('-')[0]))
+                            MetricDict[stream_name] = []
+                        #Case for "last" averaging.
+                        elif Config[stream_name][3] == 5:
+                            TotalSetTime += SetLatest(r, stream_name, MetricDict[stream_name][0])
+                            WritePostgres(p, cur, Config[stream_name][4], Config[stream_name][2], MetricDict[stream_name][1], int(MetricDict[stream_name][0].split('-')[0]))
+                            MetricDict[stream_name] = []
+                        #After performing the archiving on the previous block of data, we can now process the current entry.
+                        ProcessData(MetricDict, (entry_id, entry_fields), Config, stream_name)
+                        StreamDict[stream_name] = entry_id
+                        #Check to see if there is a gap in the data stream (the current object is more than one time interval from the latest completed).
+                        if int( entry_id.split('-')[0] ) > int( GetLatest(r, stream_name).split('-')[0] ) + 1000*Config[stream_name][5]:
+                            NewLatest = str( int( entry_id.split('-')[0] ) - 1 ) + '-0'
+                            TotalSetTime += SetLatest(r, stream_name, NewLatest)
+                            TotalSetN += 1 
+
+                except Exception as e:
+                    logging.exception(e)
+
+            if countStreams % 5000 == 0 or countStreams == totalStreams:
+                percentDone = (countStreams / totalStreams) * 100
+                logging.info("Processed {}/{} streams ({:.2f}%)".format(countStreams,totalStreams,percentDone))
+            countStreams += 1
```

Comment:
Maybe add in the archiver configuration the option to select the level of detail of the progress
```python
if countStreams % ArcConfig['logging'].get('print_every', 5000) == 0 or countStreams == totalStreams:
```
The default (if not present in the configuration `json` file) is 5000